src/main/java/com/google/api/codegen/config/GapicProductConfig.java

```diff
@@ -692,99 +709,131 @@ private static GapicProductConfig createDummyInstance(
 
   private static ImmutableMap<String, ResourceNameConfig> createResourceNameConfigs(
       DiagCollector diagCollector, ConfigProto configProto, TargetLanguage language) {
-    return createResourceNameConfigs(
-        diagCollector, configProto, null, language, ImmutableMap.of(), ImmutableMap.of(), null);
+    return createResourceNameConfigsForGapicConfigOnly(diagCollector, configProto, null, language);
   }
 
   /**
-   * Create all the ResourceNameOneofConfig from the protofile and GAPIC config, and let the GAPIC
-   * config resourceNames override the protofile resourceNames in event of clashes.
+   * Create all the ResourceNameOneofConfig from the protofile and GAPIC config. Apply the GAPIC
+   * config resourceNames' language-specific overrides.
    */
   @VisibleForTesting
   @Nullable
-  static ImmutableMap<String, ResourceNameConfig> createResourceNameConfigs(
-      DiagCollector diagCollector,
-      ConfigProto configProto,
-      @Nullable List<ProtoFile> protoFiles,
-      TargetLanguage language,
-      Map<Resource, ProtoFile> resourceDefs,
-      Map<ResourceSet, ProtoFile> resourceSetDefs,
-      ProtoParser protoParser) {
+  static ImmutableMap<String, ResourceNameConfig>
+      createResourceNameConfigsWithProtoFileAndGapicConfig(
+          DiagCollector diagCollector,
+          ConfigProto configProto,
+          @Nullable List<ProtoFile> protoFiles,
+          TargetLanguage language,
+          Map<Resource, ProtoFile> resourceDefs,
+          Map<ResourceSet, ProtoFile> resourceSetDefs,
+          ProtoParser protoParser) {
     ProtoFile file = null;
     if (protoFiles != null) {
       file = protoFiles.get(0);
     }
 
-    List<CollectionConfigProto> allCollectionConfigProtos =
-        new ArrayList<>(configProto.getCollectionsList());
-    configProto
-        .getInterfacesList()
-        .stream()
-        .forEach(i -> allCollectionConfigProtos.addAll(i.getCollectionsList()));
-    LinkedHashMap<String, SingleResourceNameConfig> singleResourceNameProtos =
-        new LinkedHashMap<>();
-    LinkedHashMap<String, FixedResourceNameConfig> fixedResourceNameProtos = new LinkedHashMap<>();
-    for (CollectionConfigProto c : allCollectionConfigProtos) {
-      if (FixedResourceNameConfig.isFixedResourceNameConfig(c.getNamePattern())) {
-        createFixedResourceNameConfig(diagCollector, c, fixedResourceNameProtos, file);
+    // Maps of fully qualified Resource names to derived configs.
+    LinkedHashMap<String, SingleResourceNameConfig>
+        fullyQualifiedSingleResourcesFromProtoFileBuilder = new LinkedHashMap<>();
```

Comment:
I don't we should call this a builder when it is a map, even though we are using it a bit like one.

Reply:
ok, calling it a `*Collector`